### ISSUE TITLE
Support for dynamic CRS, coordinate epoch

### ISSUE THREAD

**Issue by cportele** [opengeospatial/ogcapi-features#264]
Based on the discussion in the OGC Architecture Board today about the importance of being explicit about the coordinate epoch used in the geometries, if a dynamic CRS (like WGS84) is used, the CRS extension should at a minimum provide the (optional) capability to explicitly state in /collections/{collectionId} the coordinate epoch of the geometries in the collection.

Maybe it should also be possible to request coordinates in a specific epoch, if the requested CRS (query parameter `crs`) is a dynamic CRS. However, this may be too early as the underlying libraries are not yet available.

**Comment by cportele** [comment]
Here is the link to the relevant concept in the abstract spec: https://docs.opengeospatial.org/as/18-005r4/18-005r4.html#68

So the value of a `coordinateEpoch` property would be a decimal value.

Should we adds this now or wait for the emerging practice and update this in a revision? I could write a pull request.

**Comment by cportele** [comment]
Discussed in 2019-10-28 call: @pvretano will have a look. In general we should add it, if it is sufficiently clear and optional.

**Comment by jerstlouis** [comment]
What about the specification of e.g. the WGS84 realizations?
Isn't that also not specified by http://www.opengis.net/def/crs/OGC/1.3/CRS84 and required for accuracy?

**Comment by cportele** [comment]
> What about the specification of e.g. the WGS84 realizations?
> Isn't that also not specified by http://www.opengis.net/def/crs/OGC/1.3/CRS84 and required for accuracy?

CRS84 (like 4326) is based on the datum ensemble, not one of the realizations. The use of CRS84 is not wrong and should be ok in most cases, also for higher accuracy requirements, if combined with the epoch information of the data. See the slides from the OAB meeting. This is at least how I understood the discussion, but we can confirm that next week with the CRS experts at the OGC TC meeting. 

For those cases where the realization is needed, the server can state that, e.g., http://www.opengis.net/def/crs/EPSG/0/7664 is the storage CRS and support that CRS, too.

**Comment by jerstlouis** [comment]
Right, I came to learn this today.

However, it would be nice if clients and services supporting only "Core" (not the CRS extension), could still communicate both the epoch and the specific WGS84 realization, while still interoperating without either of them implementing the CRS extension.

This way neither the client nor the service really needs to care about the technicalities of dynamic CRSes, or having to specify a CRS, but they can still carry that detailed information about what exactly is the accurate CRS for the data (e.g. epoch and EPSG:9057).

In other words I guess this means the service being to state the CRS and epoch (would storage CRS be accurate for this?), without having to support the CRS extension.

**Comment by jerstlouis** [comment]
> The use of CRS84 is not wrong and should be ok in most cases, also for higher accuracy requirements, if combined with the epoch information of the data.

As I learned earlier from Even, CRS84 + an epoch is not enough for accuracy, as the difference between the first and later realizations can be as much as 2 meters.

**Comment by cportele** [comment]
I just did a Google search and according to https://confluence.qps.nl/qinsy/8.0/en/world-geodetic-system-1984-wgs84-29855173.html#WorldGeodeticSystem1984(WGS84)-WGS84realizations the shift is 0.08m since the 1997.0 realization.

**Comment by jerstlouis** [comment]
@cportele yes I believe the big shift is between the first and later realizations. The recent changes are much smaller.

**Comment by rcoup** [comment]
OSGeo/OGC background whitepaper on this issue: [Misaligned maps? High-accuracy data must become time-dependent](https://www.opengeospatial.org/blog/3045) (Aug 2019)
